extensions/ruby/src/language_servers/rubocop.rs

```diff
@@ -9,11 +11,43 @@ impl Rubocop {
         Self {}
     }
 
-    pub fn server_script_path(&mut self, worktree: &zed::Worktree) -> Result<String> {
+    pub fn language_server_binary_path(&self, worktree: &zed::Worktree) -> Result<String> {
         let path = worktree.which("rubocop").ok_or_else(|| {
             "rubocop must be installed manually. Install it with `gem install rubocop` or specify the 'binary' path to it via local settings.".to_string()
         })?;
 
         Ok(path)
     }
+
+    pub fn language_server_command(
+        &self,
+        worktree: &zed::Worktree,
+    ) -> Result<RubyLanguageServerCommand> {
+        let mut binary = None;
+        let mut args = None;
+
+        if let Some(binary_settings) =
+            LspSettings::for_worktree(Rubocop::LANGUAGE_SERVER_ID, worktree)
+                .ok()
+                .and_then(|lsp_settings| lsp_settings.binary)
+        {
+            if let Some(bin_path) = binary_settings.path {
+                binary = Some(bin_path);
+            }
+            if let Some(bin_args) = binary_settings.arguments {
+                args = Some(bin_args);
+            }
+        }
+        let command = if let Some(binary) = binary {
+            binary
+        } else {
+            self.language_server_binary_path(worktree)?
+        };
+        let args = args.unwrap_or_else(|| vec!["--lsp".into()]);
```

Comment:
My idea is to move this code into the `ruby.rs` file, use the passed `language_server_id` to pull the binary settings, and then ask the corresponding LSP for the correct arguments. But I failed to achieve that in this version. Perhaps I need another round.